### PR TITLE
Default parameters from git config

### PR DESCRIPTION
**Pull request by MartinSGill**
Implementation of #46 using new parameter types to detect specific content.

### CODE REVIEW

**Comment by rkeithhill** [comment]
Well that was quick. :-)  I'm wondering if we should maybe change the name of the input attribute.  After all, the choice and multichoice are also input.  Maybe the attribute should be called it "text" as in `<parameter type="text".../>` and then `<parameter type="author".../>` seems to make more sense to me.

BTW do you think you could change the code to avoid using git.exe.  We've had issues with git (running in a script) hanging in VSCode.  A bug with the VSCode PowerShell extension but it's not fixed yet.  Can you just parse (regex match) the data out of $home\.gitconfig?

**Comment by rkeithhill** [comment]
Agreed.

File: InvokePlaster.ps1 [hunk 1]
```diff
@@ -239,6 +239,20 @@ __________.__                   __
             $retval
         }
 
+        function GetGitConfigValue($name) {
```
Comment (rkeithhill) [inline InvokePlaster.ps1]: Oh yeah, can you move this helper function to the bottom of the file with the other helper functions?  Minor nit but would save me from having to do it.  :-)

**Comment by MartinSGill** [comment]
I agree with changing the type to 'text'. Longer term that will allow some in-built validation, if the field types are expanded, e.g. URL fields, or number fields.

File: InvokePlaster.ps1 [hunk 2]
```diff
@@ -281,6 +295,33 @@ __________.__                   __
                                 $prompt += " ($default)"
                             }
                         }
+                        # Prompt the user for text input.
+                        $value = PromptForInput $prompt $default
+                        $valueToStore = $value
+                    }
+                    'author' {
+                        # If no default, try to get a name from git config
+                        if ([String]::IsNullOrWhitespace($default)) {
```
Comment (rkeithhill) [inline InvokePlaster.ps1]: To stick with the way this script has been making this check, can we use simply `if ($default) {`.  Also, for author and email, it should allow the template author to specify a store attribute value of `encrypted`.  In this case, both of these should get the `$obscuredDefault` value and display that in the prompt.

**Pull request merged**